OpenRA.Mods.Common/Traits/Render/WithMakeAnimation.cs

```diff
@@ -20,36 +21,79 @@ public class WithMakeAnimationInfo : ITraitInfo, Requires<WithSpriteBodyInfo>
 		[Desc("Sequence name to use")]
 		[SequenceReference] public readonly string Sequence = "make";
 
+		[GrantedConditionReference]
+		[Desc("The condition to grant to self while the make animation is playing.")]
+		public readonly string Condition = null;
+
 		public object Create(ActorInitializer init) { return new WithMakeAnimation(init, this); }
 	}
 
-	public class WithMakeAnimation
+	public class WithMakeAnimation : INotifyCreated
 	{
 		readonly WithMakeAnimationInfo info;
 		readonly WithSpriteBody wsb;
+		int token = ConditionManager.InvalidConditionToken;
 
 		public WithMakeAnimation(ActorInitializer init, WithMakeAnimationInfo info)
 		{
 			this.info = info;
 			var self = init.Self;
 			wsb = self.Trait<WithSpriteBody>();
+		}
 
+		void INotifyCreated.Created(Actor self)
+		{
 			var building = self.TraitOrDefault<Building>();
 			if (building != null && !building.SkipMakeAnimation)
+				Forward(self, () => building.NotifyBuildingComplete(self));
+		}
+
+		void Forward(Actor self, Action onComplete)
+		{
+			var conditionManager = self.TraitOrDefault<ConditionManager>();
+			if (conditionManager != null && !string.IsNullOrEmpty(info.Condition) && token == ConditionManager.InvalidConditionToken)
+				token = conditionManager.GrantCondition(self, info.Condition);
+
+			wsb.PlayCustomAnimation(self, info.Sequence, () =>
 			{
-				wsb.PlayCustomAnimation(self, info.Sequence, () =>
-				{
-					building.NotifyBuildingComplete(self);
-				});
-			}
+				if (token != ConditionManager.InvalidConditionToken)
+					token = conditionManager.RevokeCondition(self, token);
+
+				// TODO: Rewrite this to use a trait notification for save game support
+				onComplete();
```

Comment:
This could be done soon by granting a condition, or, in the future, an expression variable not unlike my `health.Damage == DamageStates.critical` example: `makeState == makeState.Complete`. Other traits would just trigger when the condition is true.